### PR TITLE
Fix course cards links are duplicated

### LINKED ISSUE #2769: In the course cards, all links are duplicated.
In the course cards, all links are duplicated. 

image

While this doesn't cause any immediate problems, it does mean that the travel path is significantly longer for keyboard and screen reader users, and creates a lot of redundancy in navigating course archives. It would be better if these two elements were combined into a single link & the alt text was empty on the image.

_Originally posted by @joedolson in https://github.com/WordPress/Learn/issues/2482#issuecomment-2249050547_

### PR DESCRIPTION
**Pull request by outdoor2kode**
Fixes #2769

This PR directly adds 'aria-hidden: true' and 'tabindex: -1' to the feature images, allowing screen readers and the tab key to skip the feature image while keeping it clickable with the mouse.

**Testing:**
1. Use a screen reader to test that it does not stop on the feature image on pages where the card course template part or my course page content pattern is used.
2. Also, test that the tab key does not navigate to the feature image.
3. Test if the image is still clickable with the mouse.

### CODE REVIEW

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/index.php [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+	$metadata = wp_json_file_decode( $metadata_file, array( 'associative' => true ) );
+	$metadata['file'] = $metadata_file;
+
+	$script_handle = register_block_script_handle( $metadata, 'viewScript', 0 );
+
+	// Enqueue the assets when the card course template is on the page.
+	add_action(
+		'render_block_core/template-part',
+		function( $block_content, $block ) use ( $script_handle ) {
+			$slugs = array( 'card-course-h3', 'card-course', 'card-lesson-h3', 'card-lesson', 'card' );
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/src/card-course/index.php]: I think we need `card-lesson-plan` too

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/view.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+	 *
+	 * This function adds an `aria-hidden="true"` and `tabindex="-1"`
+	 * to the post featured image block in the card course template.
+	 */
+	function enhanceCardCourseAccessibility() {
+		document
+			.querySelectorAll( '.wporg-learn-card-grid figure.wp-block-post-featured-image' )
+			.forEach( ( featuredImg ) => {
+				featuredImg.setAttribute( 'aria-hidden', 'true' );
+				featuredImg.setAttribute( 'tabindex', '-1' );
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/src/card-course/view.js]: I think we should only be doing this to the anchor element

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/block.json [hunk 3]
```diff
@@ -0,0 +1,4 @@
+{
+	"name": "wporg-learn/card-course",
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/src/card-course/block.json]: Let's rename all the things to
```suggestion
	"name": "wporg-learn/card-featured-image-a11y",
```

**Review by adamwoodnz — COMMENTED** [review]
Nice work! Couple of comments inline.

I was going to approach this either by just removing the links from images (would have affected user experience IMO), or using our LinkWrapper block (would have required updates to make it work within a query loop), but I think this is a reasonable solution 👍

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/view.js (hunk 2 above)
Comment (adamwoodnz) [reply wp-content/themes/pub/wporg-learn-2024/src/card-course/view.js]: We could consider setting `role="presentation"` on the image, as the alt text is the same as the title and therefore it won't be adding any value.

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/block.json (hunk 3 above)
Comment (outdoor2kode) [reply wp-content/themes/pub/wporg-learn-2024/src/card-course/block.json]: caaab48

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/index.php (hunk 1 above)
Comment (outdoor2kode) [reply wp-content/themes/pub/wporg-learn-2024/src/card-course/index.php]: I don't see the card-lesson-plan, and I can see that the lesson plan archive is already working now. It doesn't work on your side?

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/view.js (hunk 2 above)
Comment (outdoor2kode) [reply wp-content/themes/pub/wporg-learn-2024/src/card-course/view.js]: >  think we should only be doing this to the anchor element

Ah, yeah, thanks! 18a01a2

> We could consider setting role="presentation" on the image, as the alt text is the same as the title and therefore it won't be adding any value.

| Before | After |
|-|-|
| Screenshot 2024-07-26 at 08 38 59 | Screenshot 2024-07-26 at 08 46 03 |

File: wp-content/themes/pub/wporg-learn-2024/src/card-course/index.php (hunk 1 above)
Comment (adamwoodnz) [reply wp-content/themes/pub/wporg-learn-2024/src/card-course/index.php]: Oh yeah I was looking at sandbox, I must have synced a file in there at some point, ignore

File: wp-content/themes/pub/wporg-learn-2024/src/card-featured-image-a11y/index.php [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+ * `wp_json_file_decode` & `register_block_script_handle.
+ */
+function init() {
+	$metadata_file = dirname( dirname( __DIR__ ) ) . '/build/card-featured-image-a11y/block.json';
+	$metadata = wp_json_file_decode( $metadata_file, array( 'associative' => true ) );
+	$metadata['file'] = $metadata_file;
+
+	$script_handle = register_block_script_handle( $metadata, 'viewScript', 0 );
+
+	// Enqueue the assets when the card course template is on the page.
```
Comment (adamwoodnz) [inline wp-content/themes/pub/wporg-learn-2024/src/card-featured-image-a11y/index.php]:
```suggestion
	// Enqueue the assets when one of the card templates is on the page.
```

Comment (outdoor2kode) [reply wp-content/themes/pub/wporg-learn-2024/src/card-featured-image-a11y/index.php]: e805a9a

**Review by adamwoodnz — APPROVED** [review]
LGTM :shipit:

**Pull request merged**